src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicHelpers.cs

```diff
@@ -38,20 +38,19 @@ internal static bool TryParse(this EndPoint endPoint, out string? host, out IPAd
     internal static unsafe IPEndPoint ToIPEndPoint(this ref QuicAddr quicAddress, AddressFamily? addressFamilyOverride = null)
     {
         // MsQuic always uses storage size as if IPv6 was used
-        Span<byte> addressBytes = new Span<byte>((byte*)Unsafe.AsPointer(ref quicAddress), Internals.SocketAddress.IPv6AddressSize);
-        return new Internals.SocketAddress(addressFamilyOverride ?? SocketAddressPal.GetAddressFamily(addressBytes), addressBytes).GetIPEndPoint();
+        Span<byte> addressBytes = new Span<byte>((byte*)Unsafe.AsPointer(ref quicAddress), SocketAddressPal.IPv6AddressSize);
```

Comment:
This looks like a GC hole (both before and with this PR).  Is quicAddress guaranteed to be pinned or otherwise immovable?

Reply:
When we get it, it is allocated by msquic from native memory. e.g. it would be immovable at that point. I can check all the references where we get it.

Reply:
I check and AFAIK we call it only on memory. Perhaps we can make it  `QuicAddr*` to make it more obvious? It is really just generated cast to C structure.

Reply:
> Perhaps we can make it QuicAddr* to make it more obvious?

That would be better IMO, yes.  Any use of AsPointer is a red flag to be investigated; while it does have its valid uses, they're rare. If we can remove it from here, one less thing to think about in code reviews like this.

Reply:
updated. cc @ManickaP  for visibility. For the logs, we should be able to log address without allocating IPEndPoint. Perhaps something to improve in 9.0...